.github/workflows/release-pr.yml

```diff
@@ -0,0 +1,92 @@
+name: Post-Release PR
+
+permissions:
+  # To create a PR
+  contents: write
+  pull-requests: write
+
+on:
+  workflow_call:
+    inputs:
+      ref:
+        required: true
+        description: "GitHub ref to use"
+        type: string
+      version:
+        required: true
+        description: "Version to produce"
+        type: string
+      branch_from_ref:
+        required: false
+        description: "Commit to branch from, if not the tag"
+        type: string
+      next-version:
+        required: true
+        description: "Version to bump files to"
+        type: string
+      release-notes:
+        required: true
+        description: "Release notes to publish"
+        type: string
+      queue-merge:
+        required: false
+        default: false
+        description: "Whether to queue the release for immediate merge"
+        type: boolean
+
+jobs:
+  version-bump:
+    name: version bump
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{ inputs.ref }}
+      - uses: actions/setup-go@v3
+        with:
+          go-version: 1.19.x
+      - name: Create PR
+        env:
+          PULUMI_VERSION: ${{ inputs.version }}
+          NEXT_VERSION: ${{ inputs.next-version }}
+          RELEASE_NOTES: ${{ inputs.release-notes }}
+          QUEUE_MERGE: ${{ inputs.queue-merge }}
+          GITHUB_TOKEN: ${{ secrets.PULUMI_BOT_TOKEN }}
+        run: |
+          set -euo pipefail
+          git switch --create "automation/release-v${PULUMI_VERSION}-v${NEXT_VERSION}"
+
+          echo -en "# Changelog\n\n${RELEASE_NOTES}\n\n$(tail -n+3 CHANGELOG.md)" > ./CHANGELOG.md
+
+          ./.github/scripts/update-versions "${NEXT_VERSION}"
+
+          git config user.name github-actions
+          git config user.email github-actions@github.com
+
+          rm ./changelog/pending/*.yaml || true
+          git add -A
+
+          git commit -m "Prepare for next release (v${NEXT_VERSION})"
+
+          # Update go module dependencies
+          (
+            cd pkg
+            go get -u "github.com/pulumi/pulumi/sdk/v3@v${PULUMI_VERSION}"
+          )
+          make tidy
+          git add -A
+          git commit -m "Update dependencies"
+
+          # Publish pkg module on another tag.
+          git tag "pkg/v${PULUMI_VERSION}"
+          git push origin "pkg/v${PULUMI_VERSION}"
```

Comment:
Curious how this works in go tooling. If we push pkg/v3.100.0 tag, but on a branch that is not the main branch of a repository, will `go` tooling consider that published, e.g. resolve go.mod references to v3.100.0 to that tag even if it is not on master? Checking this out quickly.

Reply:
Checked that! Never mind, all good. `go mod` does not resolve a new tag until it's on main branch.